app/controllers/candidate_interface/application_form_controller.rb

```diff
@@ -82,5 +93,11 @@ def further_information_params
       )
         .transform_values(&:strip)
     end
+
+    def build_previous_application_form
+      return if params['candidate_id']&.to_i != current_candidate.id
```

Comment:
If I understand it correctly, is this meant to avoid people seeing other people's applications? It's doesn't work for that, since they can change the ID in the URL to their own, and this will not `return`.

Luckily, the `current_candidate.application_forms.find()` is the correct approach so we don't need this extra check here.